app/src/main/kotlin/com/wire/android/migration/MigrationManager.kt

```diff
@@ -386,5 +406,10 @@ fun Data.getMigrationFailure(): MigrationData.Result.Failure = when (this.getStr
         this.getString(MigrationData.Result.Failure.KEY_FAILURE_ERROR_CODE) ?: ""
     )
 
-    else -> MigrationData.Result.Failure.NoNetwork
+    MigrationData.Result.Failure.KEY_MIGRATION_EXCEPTION -> this.keyValueMap[MigrationData.Result.Failure.KEY_MIGRATION_EXCEPTION]?.let {
```

Comment:
The idea of this `when` was to check the `MigrationData.Result.Failure.KEY_FAILURE_TYPE` and then get the required fields like `KEY_MIGRATION_EXCEPTION` depending on the type of failure